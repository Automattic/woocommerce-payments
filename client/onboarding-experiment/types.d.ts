export interface Country {
	key: string;
	name: string;
	types: BusinessType[];
}

export interface BusinessType {
	key: string;
	name: string;
	description: string;
	structures: BusinessStructure[];
}

export interface BusinessStructure {
	key: string;
	name: string;
}

export interface OnboardingProps {
	country: string;
	type: string;
	structure?: string;
}
