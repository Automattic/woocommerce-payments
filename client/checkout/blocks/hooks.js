/**
 * External dependencies
 */
import { useEffect, useState } from '@wordpress/element';

/**
 * Internal dependencies
 */
import confirmCardPayment from './confirm-card-payment.js';
import {
	getFingerprint,
	FINGERPRINT_GENERIC_ERROR,
} from '../utils/fingerprint.js';

export const usePaymentCompleteHandler = (
	api,
	stripe,
	elements,
	onCheckoutSuccess,
	emitResponse,
	shouldSavePayment
) => {
	// Once the server has completed payment processing, confirm the intent of necessary.
	useEffect(
		() =>
			onCheckoutSuccess( ( { processingResponse: { paymentDetails } } ) =>
				confirmCardPayment(
					api,
					paymentDetails,
					emitResponse,
					shouldSavePayment
				)
			),
		// not sure if we need to disable this, but kept it as-is to ensure nothing breaks. Please consider passing all the deps.
		// eslint-disable-next-line react-hooks/exhaustive-deps
		[ elements, stripe, api, shouldSavePayment ]
	);
};

/**
 * Handles the Block Checkout onCheckoutFail event.
 *
 * Displays the error message returned from server in the paymentDetails object in the PAYMENTS notice context container.
 *
 * @param {*} onCheckoutFail The onCheckoutFail event.
 * @param {*} emitResponse   Various helpers for usage with observer.
 */
export const usePaymentFailHandler = ( onCheckoutFail, emitResponse ) => {
	useEffect(
		() =>
			onCheckoutFail( ( { processingResponse: { paymentDetails } } ) => {
				return {
					type: 'failure',
					message: paymentDetails.errorMessage,
					messageContext: emitResponse.noticeContexts.PAYMENTS,
				};
			} ),
		[ onCheckoutFail, emitResponse.noticeContexts.PAYMENTS ]
	);
};

export const useFingerprint = () => {
	const [ fingerprint, setFingerprint ] = useState( '' );
	const [ error, setError ] = useState( null );

	useEffect( () => {
		( async () => {
			try {
				const { visitorId } = await getFingerprint();
				setFingerprint( visitorId );
			} catch ( err ) {
				setError(
					err.message ? err.message : FINGERPRINT_GENERIC_ERROR
				);
			}
		} )();
	}, [] );

	return [ fingerprint, error ];
};
