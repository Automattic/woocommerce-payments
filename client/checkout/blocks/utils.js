/**
 * External dependencies
 */
import { useDispatch, useSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import { WC_STORE_CART } from 'wcpay/checkout/constants';

/**
 *
 * Custom React hook that provides customer data and related functions for managing customer information.
 * The hook retrieves customer data from the WC_STORE_CART selector and dispatches actions to modify billing and shipping addresses.
 *
 * @return {Object} An object containing customer data and functions for managing customer information.
 */
export const useCustomerData = () => {
	const customerData = useSelect( ( select ) =>
		select( WC_STORE_CART ).getCustomerData()
	);
	const {
		setShippingAddress,
		setBillingData,
		setBillingAddress,
	} = useDispatch( WC_STORE_CART );

	return {
		// Backward compatibility billingData/billingAddress
		billingAddress: customerData.billingAddress || customerData.billingData,
		// Backward compatibility setBillingData/setBillingAddress
		setBillingAddress: setBillingAddress || setBillingData,
		setShippingAddress,
	};
};
