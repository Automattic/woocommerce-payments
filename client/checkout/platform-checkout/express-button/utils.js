/* global wcpayWooPayExpressParams */

/**
 * Retrieves a configuration value.
 *
 * @param {string} name The name of the config parameter.
 * @return {*}         The value of the parameter of null.
 */
export const getWooPayExpressData = ( name ) => {
	if ( 'undefined' === typeof wcpayWooPayExpressParams ) {
		return undefined;
	}

	return wcpayWooPayExpressParams?.[ name ];
};
