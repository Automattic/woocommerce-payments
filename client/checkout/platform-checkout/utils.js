// Waits for the element to exist as in the Blocks checkout, sometimes the field is not immediately available.
export const getTargetElement = ( selector ) => {
	if ( ! selector ) {
		return null;
	}
	return new Promise( ( resolve ) => {
		if ( document.querySelector( selector ) ) {
			return resolve( document.querySelector( selector ) );
		}

		const checkoutBlock = document.querySelector(
			'[data-block-name="woocommerce/checkout"]'
		);

		if ( ! checkoutBlock ) {
			return resolve( null );
		}

		const observer = new MutationObserver( ( mutationList, obs ) => {
			if ( document.querySelector( selector ) ) {
				resolve( document.querySelector( selector ) );
				obs.disconnect();
			}
		} );

		observer.observe( checkoutBlock, {
			childList: true,
			subtree: true,
		} );
	} );
};

export const validateEmail = ( value ) => {
	/* Borrowed from WooCommerce checkout.js with a slight tweak to add `{2,}` to the end and make the TLD at least 2 characters. */
	/* eslint-disable */
	const pattern = new RegExp(
		/^([a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+(\.[a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+)*|"((([ \t]*\r\n)?[ \t]+)?([\x01-\x08\x0b\x0c\x0e-\x1f\x7f\x21\x23-\x5b\x5d-\x7e\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|\\[\x01-\x09\x0b\x0c\x0d-\x7f\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))*(([ \t]*\r\n)?[ \t]+)?")@(([a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])\.)+([a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[0-9a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]){2,}\.?$/i
	);
	/* eslint-enable */
	return pattern.test( value );
};
