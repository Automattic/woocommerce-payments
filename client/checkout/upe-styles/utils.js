/**
 * External dependencies
 */
import tinycolor from 'tinycolor2';

/**
 * Generates hover colors from a background color and a text color.
 *
 * @param {string}  backgroundColor Background color, Any format accepted by tinyColor library
 * @param {string}  color Text color, any format accepted by tinyColor library
 * @return {Object} Object with new background color and text color.
 */
export const generateHoverColors = ( backgroundColor, color ) => {
	const hoverColors = {
		backgroundColor,
		color,
	};

	const tinyBackgroundColor = tinycolor( backgroundColor );
	const tinyTextColor = tinycolor( color );

	// If colors are not valid return empty strings.
	if ( ! tinyBackgroundColor.isValid() || ! tinyTextColor.isValid() ) {
		return {
			backgroundColor: '',
			color: '',
		};
	}

	// Darken if brightness > 50 (Storefront Button 51 ), else lighten
	const newBackgroundColor =
		tinyBackgroundColor.getBrightness() > 50
			? tinycolor( tinyBackgroundColor ).darken( 7 )
			: tinycolor( tinyBackgroundColor ).lighten( 7 );

	// Returns provided color if readable, otherwise black or white.
	const mostReadableColor = tinycolor.mostReadable(
		newBackgroundColor,
		[ tinyTextColor ],
		{ includeFallbackColors: true }
	);

	hoverColors.backgroundColor = newBackgroundColor.toRgbString();
	hoverColors.color = mostReadableColor.toRgbString();

	return hoverColors;
};

/**
 * Generates hover rules for UPE using a set of appearance rules as a basis.
 *
 * @param {Object}  baseRules UPE appearance rules to use as a base to generate hover colors
 * @return {Object} Object with generated hover rules.
 */
export const generateHoverRules = ( baseRules ) => {
	const hoverRules = Object.assign( {}, baseRules );

	// If there are no colors, return the same rules as we can not generate hover colors.
	if ( ! baseRules.backgroundColor || ! baseRules.color ) {
		return baseRules;
	}

	const hoverColors = generateHoverColors(
		baseRules.backgroundColor,
		baseRules.color
	);

	hoverRules.backgroundColor = hoverColors.backgroundColor;
	hoverRules.color = hoverColors.color;

	return hoverRules;
};

/**
 * Generates outline style for UPE using outline width, style and color.
 * UPE does not accept the individual properties, we need to concat them.
 *
 * @param {string}  outlineWidth Outline width from computed styles.
 * @param {string}  outlineStyle Outline width from computed styles.
 * @param {string}  outlineColor Outline width from computed styles.
 * @return {string} Object with generated hover rules.
 */
export const generateOutlineStyle = (
	outlineWidth,
	outlineStyle = 'solid',
	outlineColor
) => {
	return outlineWidth && outlineColor
		? [ outlineWidth, outlineStyle, outlineColor ].join( ' ' )
		: '';
};

/**
 * Converts CSS property from dashed format to camel case.
 *
 * @param {string} string CSS property.
 * @return {string} Camel case string.
 */
export const dashedToCamelCase = ( string ) => {
	return string.replace( /-([a-z])/g, function ( g ) {
		return g[ 1 ].toUpperCase();
	} );
};

/**
 * Converts rgba to rgb format, since Stripe Appearances API does not accept rgba format for background.
 *
 * @param {string} color CSS color value.
 * @return {string} Accepted CSS color value.
 */
export const maybeConvertRGBAtoRGB = ( color ) => {
	if ( color.match( /^rgba/ ) ) {
		const colorParts = color.match(
			/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/
		);
		color = `rgb(${ colorParts[ 1 ] }, ${ colorParts[ 2 ] }, ${ colorParts[ 3 ] })`;
	}
	return color;
};

/**
 * Searches through array of CSS selectors and returns first visible background color.
 *
 * @param {Array} selectors List of CSS selectors to check.
 * @return {string} CSS color value.
 */
export const getBackgroundColor = ( selectors ) => {
	const defaultColor = '#ffffff';
	let color = null;
	let i = 0;
	while ( ! color && i < selectors.length ) {
		const bgColor = window.getComputedStyle(
			document.querySelector( selectors[ i ] )
		).backgroundColor;
		if ( bgColor.match( /^rgba/ ) ) {
			const colorParts = bgColor.match(
				/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/
			);
			// Check if color is transparent.
			if ( parseInt( colorParts[ 5 ], 10 ) > 0 ) {
				color = bgColor;
			}
		} else {
			color = bgColor;
		}
		i++;
	}
	return color || defaultColor;
};

/**
 * Determines whether background color is light or dark.
 *
 * @param {string} color CSS color value.
 * @return {boolean} True, if background is light; false, if background is dark.
 */
export const isColorLight = ( color ) => {
	let r, g, b;
	if ( color.match( /^rgb/ ) ) {
		color = color.match(
			/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/
		);
		[ r, g, b ] = color.slice( 1 );
	} else {
		// Convert hex format to rgb
		color = +(
			'0x' + color.slice( 1 ).replace( color.length < 5 && /./g, '$&$&' )
		);

		// eslint-disable-next-line no-bitwise
		r = color >> 16;
		// eslint-disable-next-line no-bitwise
		g = ( color >> 8 ) & 255;
		// eslint-disable-next-line no-bitwise
		b = color & 255;
	}

	// HSP equation from http://alienryderflex.com/hsp.html
	const hsp = Math.sqrt(
		0.299 * ( r * r ) + 0.587 * ( g * g ) + 0.114 * ( b * b )
	);

	// Using the HSP value, determine whether the color is light or dark
	return hsp > 127.5;
};
