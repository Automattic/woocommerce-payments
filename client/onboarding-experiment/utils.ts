/**
 * Internal dependencies
 */
import businessTypeStrings from 'onboarding-experiment/translations/types';
import businessStructureStrings from 'onboarding-experiment/translations/structures';
import businessTypeDescriptionStrings from 'onboarding-experiment/translations/descriptions';

type CountriesData = Country[];

export const getBusinessTypes = (): CountriesData => {
	const data = wcpaySettings?.onboardingFieldsData?.business_types;

	return (
		( data || [] )
			.map( ( country ) => ( {
				key: country.key,
				name: country.name,
				types: country.types.map( ( type ) => ( {
					key: type.key,
					name: businessTypeStrings[ type.key ],
					description: businessTypeDescriptionStrings[ country.key ]
						? businessTypeDescriptionStrings[ country.key ][
								type.key
						  ]
						: businessTypeDescriptionStrings.generic[ type.key ],
					structures: type.structures.map( ( structure ) => ( {
						key: structure.key,
						name:
							businessStructureStrings[ country.key ][
								structure.key
							],
					} ) ),
				} ) ),
			} ) )
			.sort( ( a, b ) => a.name.localeCompare( b.name ) ) || []
	);
};
