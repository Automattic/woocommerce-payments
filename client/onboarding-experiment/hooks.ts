/**
 * External dependencies
 */
import { addQueryArgs } from '@wordpress/url';
import { __ } from '@wordpress/i18n';
import { NAMESPACE } from 'data/constants';

/**
 * Internal dependencies
 */
import { useApiFetch } from 'hooks/use-api-fetch';
import { OnboardingProps } from './types';

type VerificationInfoData = string[];

interface VerificationInfoReturn {
	requiredInfo: VerificationInfoData;
	isLoading: boolean;
}

export const useRequiredVerificationInfo = (
	query: OnboardingProps
): VerificationInfoReturn => {
	const path = addQueryArgs(
		`${ NAMESPACE }/onboarding/required_verification_information`,
		query
	);
	const { data, isLoading } = useApiFetch< VerificationInfoData >( {
		path,
		errorMessage: __(
			'Error retrieving required verification information',
			'woocommerce-payments'
		),
	} );

	return {
		requiredInfo: data || [],
		isLoading,
	};
};
