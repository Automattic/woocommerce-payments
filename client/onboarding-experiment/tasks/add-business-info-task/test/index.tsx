/**
 * External dependencies
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import { mocked } from 'ts-jest/utils';

/**
 * Internal dependencies
 */
import AddBusinessInfoTask from '../';
import { useBusinessTypes } from 'onboarding-experiment/hooks';

declare const global: {
	wcpaySettings: {
		connect: {
			country: string;
		};
	};
};

jest.mock( 'onboarding-experiment/hooks', () => ( {
	useBusinessTypes: jest.fn(),
} ) );

const countries = [
	{
		key: 'US',
		name: 'United States',
		types: [
			{
				key: 'individual',
				name: 'Individual',
				description: 'Individual',
				structures: [],
			},
			{
				key: 'company',
				name: 'Company',
				description: 'Company',
				structures: [
					{
						key: 'sole_proprietorship',
						name: 'Sole proprietorship',
					},
					{
						key: 'single_member_llc',
						name: 'Single member LLC',
					},
				],
			},
		],
	},
	{
		key: 'FR',
		name: 'France',
		types: [
			{
				key: 'individual',
				name: 'Individual',
				description: 'Individual',
				structures: [],
			},
			{
				key: 'company',
				name: 'Company',
				description: 'Company',
				structures: [],
			},
			{
				key: 'non_profit',
				name: 'Non-profit',
				description: 'Non-profit',
				structures: [],
			},
		],
	},
];

const renderTask = () =>
	render( <AddBusinessInfoTask onChange={ jest.fn() } /> );

describe( 'AddBusinessInfoTask', () => {
	beforeEach( () => {
		jest.clearAllMocks();
		global.wcpaySettings = {
			connect: {
				country: 'US',
			},
		};
	} );

	it( 'shows a loadable block', () => {
		mocked( useBusinessTypes ).mockReturnValue( {
			countries: [],
			isLoading: true,
		} );

		const { container: task } = renderTask();
		expect( task ).toMatchSnapshot();
	} );

	it( 'shows the form', () => {
		mocked( useBusinessTypes ).mockReturnValue( {
			countries,
			isLoading: false,
		} );

		const { container: task } = renderTask();
		expect( task ).toMatchSnapshot();
	} );

	it( 'shows business type and structure from a selected country', () => {
		global.wcpaySettings = {
			connect: {
				country: 'FR',
			},
		};
		mocked( useBusinessTypes ).mockReturnValue( {
			countries,
			isLoading: false,
		} );

		const { container: task } = renderTask();

		user.click( screen.getByRole( 'button', { name: /country/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /united states/i,
			} )
		);

		user.click( screen.getByRole( 'button', { name: /business type/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /company/i,
			} )
		);

		user.click(
			screen.getByRole( 'button', { name: /business structure/i } )
		);
		user.click(
			screen.getByRole( 'option', {
				name: /single member llc/i,
			} )
		);

		expect( task ).toMatchSnapshot();
	} );

	it( 'hides the structure when no structures are available', () => {
		global.wcpaySettings = {
			connect: {
				country: 'FR',
			},
		};
		mocked( useBusinessTypes ).mockReturnValue( {
			countries,
			isLoading: false,
		} );

		const { container: task } = renderTask();

		user.click( screen.getByRole( 'button', { name: /country/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /united states/i,
			} )
		);

		user.click( screen.getByRole( 'button', { name: /business type/i } ) );
		user.click(
			screen.getByRole( 'option', {
				name: /individual/i,
			} )
		);

		expect( task ).toMatchSnapshot();
		expect(
			screen.queryByRole( 'button', { name: /business structure/i } )
		).not.toBeInTheDocument();
	} );
} );
