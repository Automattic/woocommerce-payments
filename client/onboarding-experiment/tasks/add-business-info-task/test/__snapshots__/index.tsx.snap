// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBusinessInfoTask hides the structure when no structures are available 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable deposits.
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-23-toggle-button"
          id="downshift-23-label"
        >
          Country
        </label>
        <button
          aria-describedby="Currently selected: United States"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Country"
          class="components-button components-custom-select-control__button"
          id="downshift-23-toggle-button"
          type="button"
        >
          <span
            class="components-custom-select-control__button-value"
          >
            United States
          </span>
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-23-label"
          class="components-custom-select-control__menu"
          id="downshift-23-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <p
        class="complete-business-info-task__description"
      >
        The primary country where your business operates
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-25-toggle-button"
          id="downshift-25-label"
        >
          Business type
        </label>
        <button
          aria-describedby="Currently selected: Individual"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business type"
          class="components-button components-custom-select-control__button"
          id="downshift-25-toggle-button"
          type="button"
        >
          <span
            class="components-custom-select-control__button-value"
          >
            Individual
          </span>
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-25-label"
          class="components-custom-select-control__menu"
          id="downshift-25-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
    </div>
  </li>
</div>
`;

exports[`AddBusinessInfoTask shows a loadable block 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable deposits.
      </p>
      <span
        aria-busy="true"
        class="is-loadable-placeholder is-block"
      >
        <p
          style="line-height: 4;"
        >
          Block placeholder
        </p>
      </span>
      <span
        aria-busy="true"
        class="is-loadable-placeholder is-block"
      >
        <p
          style="line-height: 4;"
        >
          Block placeholder
        </p>
      </span>
    </div>
  </li>
</div>
`;

exports[`AddBusinessInfoTask shows business type and structure from a selected country 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable deposits.
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-3-toggle-button"
          id="downshift-3-label"
        >
          Country
        </label>
        <button
          aria-describedby="Currently selected: United States"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Country"
          class="components-button components-custom-select-control__button"
          id="downshift-3-toggle-button"
          type="button"
        >
          <span
            class="components-custom-select-control__button-value"
          >
            United States
          </span>
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-3-label"
          class="components-custom-select-control__menu"
          id="downshift-3-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <p
        class="complete-business-info-task__description"
      >
        The primary country where your business operates
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-5-toggle-button"
          id="downshift-5-label"
        >
          Business type
        </label>
        <button
          aria-describedby="Currently selected: Company"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business type"
          class="components-button components-custom-select-control__button"
          id="downshift-5-toggle-button"
          type="button"
        >
          <span
            class="components-custom-select-control__button-value"
          >
            Company
          </span>
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-5-label"
          class="components-custom-select-control__menu"
          id="downshift-5-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-16-toggle-button"
          id="downshift-16-label"
        >
          Business Structure
        </label>
        <button
          aria-describedby="Currently selected: Single member LLC"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business Structure"
          class="components-button components-custom-select-control__button"
          id="downshift-16-toggle-button"
          type="button"
        >
          <span
            class="components-custom-select-control__button-value"
          >
            Single member LLC
          </span>
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-16-label"
          class="components-custom-select-control__menu"
          id="downshift-16-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
    </div>
  </li>
</div>
`;

exports[`AddBusinessInfoTask shows the form 1`] = `
<div>
  <li
    class="wcpay-wizard-task complete-business-info-task"
  >
    <div
      class="wcpay-wizard-task__top-border"
    />
    <div
      class="wcpay-wizard-task__headline"
      tabindex="-1"
    >
      <div
        class="wcpay-wizard-task__icon-wrapper"
      >
        <div
          class="wcpay-wizard-task__icon-text"
        >
          1
        </div>
        <svg
          aria-hidden="true"
          class="wcpay-wizard-task__icon-checkmark"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.7 7.1l-6.3 8.5-3.3-2.5-.9 1.2 4.5 3.4L17.9 8z"
          />
        </svg>
      </div>
      <span
        class="wcpay-wizard-task__title"
      >
        Tell us more about your business
      </span>
    </div>
    <div
      class="wcpay-wizard-task__body"
    >
      <p
        class="complete-business-info-task__subheading"
      >
        Preview the details we may require to verify your business and enable deposits.
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-0-toggle-button"
          id="downshift-0-label"
        >
          Country
        </label>
        <button
          aria-describedby="Currently selected: United States"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Country"
          class="components-button components-custom-select-control__button"
          id="downshift-0-toggle-button"
          type="button"
        >
          <span
            class="components-custom-select-control__button-value"
          >
            United States
          </span>
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-0-label"
          class="components-custom-select-control__menu"
          id="downshift-0-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
      <p
        class="complete-business-info-task__description"
      >
        The primary country where your business operates
      </p>
      <div
        class="wcpay components-custom-select-control"
      >
        <label
          class="components-custom-select-control__label"
          for="downshift-2-toggle-button"
          id="downshift-2-label"
        >
          Business type
        </label>
        <button
          aria-describedby="No selection"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-label="Business type"
          class="components-button components-custom-select-control__button placeholder"
          id="downshift-2-toggle-button"
          type="button"
        >
          <span
            class="components-custom-select-control__button-value"
          >
            What type of business do you run?
          </span>
          <svg
            aria-hidden="true"
            class="components-custom-select-control__button-icon"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
            />
          </svg>
        </button>
        <ul
          aria-hidden="true"
          aria-labelledby="downshift-2-label"
          class="components-custom-select-control__menu"
          id="downshift-2-menu"
          role="listbox"
          tabindex="-1"
        />
      </div>
    </div>
  </li>
</div>
`;
